### PR TITLE
Profiles: ENS Search Sheet improvements

### PR DESCRIPTION
**Pull request by benisgold**
Fixes TEAM2-137

### Problems
- ens search often takes a super long time on app startup for certain phones (anywhere from 20-30 seconds on my Samsung Galaxy) due to general/unrelated performance issues
- the ens search loading indicator doesn't appear at all on android devices, which makes it appear that ens search isn't working at all considering how long it takes

### Solutions
- parallelized some async operations in `useENSSearch`, bringing the search time down to about 10 seconds on app startup
- fix MaskElement issue that was causing the missing search indicator on android devices (see https://github.com/react-native-masked-view/masked-view/issues/132)

## PoW (screenshots / screen recordings)
https://user-images.githubusercontent.com/15272675/174412921-adfe28e8-d087-40bc-a244-e8cf613b8b53.mp4

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/hooks/useENSSearch.ts [hunk 1]
```diff
@@ -42,9 +55,17 @@ export default function useENSSearch({
...
     }
-
-    const isAvailable = await getAvailable(name);
-    const rentPrice = await getRentPrice(name, duration);
+    const [
+      isAvailable,
+      rentPrice,
+      registrationDate,
+      nameExpires,
+    ] = await promiseUtils.PromiseAllWithFails([
```
Comment (jxom) [inline src/hooks/useENSSearch.ts:63]: I believe this will also swallow errors – are we sure about that?

Comment (benisgold) [reply src/hooks/useENSSearch.ts]: i dont think it makes a difference from the user perspective but yeah looking back its probably not a good idea to blanket catch errors. i pushed a change that catches and rethrows the errors from the individual promises if applicable/relevant. 

if you think thats too messy looking, i could just break this up into two Promise.alls - one for availablility/rent price, the other for registration/expiration date (would only get called if name is unavailable).

File: src/hooks/useENSSearch.ts [hunk 2]
```diff
@@ -43,8 +56,23 @@ export default function useENSSearch({
...
+      registrationDate,
+      nameExpires,
+    ] = await Promise.all([
+      getAvailable(name, contract).catch(e => (errors[0] = e)),
+      getRentPrice(name, duration, contract).catch(e => (errors[1] = e)),
+      fetchRegistrationDate(name + ENS_DOMAIN).catch(e => (errors[2] = e)),
+      getNameExpires(name).catch(e => (errors[3] = e)),
+    ]);
+
+    if (errors[0]) throw errors[0];
```
Comment (jxom) [inline src/hooks/useENSSearch.ts:73]: can we not just do it the original way, but place the `getAvailable` and `getRentPrice` in a `Promise.all`?

File: src/hooks/useENSSearch.ts [hunk 3]
```diff
@@ -61,26 +89,30 @@ export default function useENSSearch({
         valid: true,
       };
     } else {
-      // we need the expiration and registration date when is not available
-      const registrationDate = await fetchRegistrationDate(name + ENS_DOMAIN);
-      const nameExpires = await getNameExpires(name);
+      if (errors[2]) throw errors[2];
```
Comment (jxom) [inline src/hooks/useENSSearch.ts:92]: can we not just do it the original way, but place the `fetchRegistrationDate` and `getNameExpires` in a `Promise.all`?

**Review by jxom — APPROVED** [review]

**Review by estebanmino — APPROVED** [review]
🔥

**Pull request merged**